server/controllers/stock/import.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Stock Import Module
+ *
+ * This module is responsible of handling the import of stock
+ * and related stock quantities
+ */
+const path = require('path');
+const uuid = require('uuid');
+const moment = require('moment');
+
+const db = require('../../lib/db');
+const util = require('../../lib/util');
+const BadRequest = require('../../lib/errors/BadRequest');
+
+exports.downloadTemplate = downloadTemplate;
+exports.importStock = importStock;
+
+/**
+ * @method downloadTemplate
+ *
+ * @description send to the client the template file for stock import
+*/
+function downloadTemplate(req, res, next) {
+  try {
+    const file = path.join(__dirname, '../../resources/templates/import-stock-template.csv');
+    res.download(file);
+  } catch (error) {
+    next(error);
+  }
+}
+
+/**
+ * @method importStock
+ *
+ * @description this method allow to do an import of stock and their lots
+ */
+function importStock(req, res, next) {
+  if (!req.files || req.files.length === 0) {
+    const errorDescription = 'Expected at least one file upload but did not receive any files.';
+    const errorDetails = new BadRequest(errorDescription, 'ERRORS.MISSING_UPLOAD_FILES');
+    next(errorDetails);
+    return;
+  }
+
+  let query;
+  let queryParams;
+
+  const filePath = req.files[0].path;
+  const depotUuid = db.bid(req.body.depot_uuid);
+  const documentUuid = db.bid(uuid.v4());
```

Comment:
You use use [util.uuid()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/lib/util.js#L248) to generate your UUIDs.